src/solve.jl

```diff
@@ -119,6 +119,31 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
     return solve!(sys_new, copy(options), opC, opD)
 end
 
+
+"""
+get_necessary_vars(HS::HybridSystem):: Vector{Int64}
+
+Function to get only variables which are used for guard and invariant constraints
+
+### Input
+
+- `HS`  -- hybrid system
+"""
+function get_necessary_vars(HS::HybridSystem):: Vector{Int64}
+   vars = Vector{Int64}()
+   for mode in states(HS)
+       for constraint in constraints_list(HS.modes[mode].X)
+           vars = unique(vcat(vars, find(constraint.a)))
```

Comment:
compatibility issue: consider using `findall(x -> x != 0, constraint.a)` or `findall(!iszero, constraint.a)` instead of `find(constraint.a)`